Fix the OpenJDK11 build instructions for macOS

SueChaplain opened the pull request:
Add missing dependencies and confirguration steps.

[ci-skip]

Closes: #3971

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

SueChaplain commented:
@babsingh - please can you double check my changes? Thanks!

babsingh commented on doc/build-instructions/Build_Instructions_V11.md:
```diff
@@ -469,6 +479,13 @@ tar -xzf freemarker.tgz freemarker-2.3.8/lib/freemarker.jar --strip=2
 rm -f freemarker.tgz
```
in the above `tar` cmd, `--strip=2` needs to be changed to `--strip-components=2`.

SueChaplain replied:
Will change.

babsingh commented on doc/build-instructions/Build_Instructions_V11.md:
````diff
@@ -492,13 +509,15 @@ When you have all the source files that you need, run the configure script, whic
 
 ```
 bash configure --with-freemarker-jar=/<my_home_dir>/freemarker.jar \
+               --with-freetype=/usr/local/Cellar/freetype/2.9.1/ \
````
`--with-freetype` is not required by default. it should be included as an optional step similar to **Non-compressed references support**.

SueChaplain replied:
For clarity, is freetype a definite pre-req? If not, in what situation would a user need to include the --with-freetype option? Answers will help me phrase the note. Thanks

babsingh replied:
Yes, `freetype` a definite pre-req. `--with-freetype` can be used in situations where `bash configure` is unable to auto-detect `freetype`.

SueChaplain replied:
Thanks. Note added.

babsingh commented on doc/build-instructions/Build_Instructions_V11.md:
```diff
@@ -457,10 +457,20 @@ You must install a number of software dependencies to create a suitable build en
+The following dependencies can be installed by using [Homebrew](https://brew.sh/):
+
+- [autoconf 2.6.9](https://formulae.brew.sh/formula/autoconf)
+- [bash 4.4.23](https://formulae.brew.sh/formula/bash)
```
I don't think `bash 4` is required for JDK11. There is no harm in using `bash 4`. `bash 4` can be optional. if `bash 4` is used, then the following steps should be mentioned:

Bash version 4 is installed to `/usr/local/bin/bash`. To prevent problems during the build process, make Bash v4 your default shell by typing the following commands:

```
# Find the <CURRENT_SHELL> for <USERNAME>
dscl . -read <USERNAME> UserShell

# Change the shell to Bash version 4 for <USERNAME>
dscl . -change <USERNAME> UserShell <CURRENT_SHELL> /usr/local/bin/bash

# Verify that the shell has been changed
dscl . -read <USERNAME> UserShell
```

SueChaplain commented:
@babsingh - think I have addressed all your comments now. 

I also added in OpenSSL support now that this work is in place for OpenJDK 11. Probably should have done this in a separate PR, but hope you are ok with it.

SueChaplain commented:
Yes, absolutely. Thanks @babsingh ... I've amended the commit message.
@DanHeidinga are you able to merge please?
